crates/lib/src/qpu/api.rs

```diff
@@ -257,38 +256,40 @@ impl ConnectionStrategy {
         client: &Qcs,
         quantum_processor_id: Option<&str>,
     ) -> Result<ControllerClient<RefreshService<Channel, ClientConfiguration>>, QpuApiError> {
-        let address = match self {
-            Self::EndpointId(endpoint_id) => {
+        let address = match self.connection_strategy() {
+            ConnectionStrategy::EndpointId(endpoint_id) => {
                 let endpoint = get_endpoint(&client.get_openapi_client(), endpoint_id).await?;
                 endpoint
                     .addresses
                     .grpc
                     .ok_or_else(|| QpuApiError::EndpointNotFound(endpoint_id.into()))?
             }
-            Self::Gateway => {
+            ConnectionStrategy::Gateway => {
                 self.get_gateway_address(
                     quantum_processor_id.ok_or(QpuApiError::MissingQpuId)?,
                     client,
                 )
                 .await?
             }
-            Self::DirectAccess => {
+            ConnectionStrategy::DirectAccess => {
                 self.get_default_endpoint_address(
                     quantum_processor_id.ok_or(QpuApiError::MissingQpuId)?,
                     client,
                 )
                 .await?
             }
         };
-        Self::grpc_address_to_client(&address, client)
+        self.grpc_address_to_client(&address, client)
     }
 
     fn grpc_address_to_client(
+        &self,
         address: &str,
         client: &Qcs,
     ) -> Result<ControllerClient<RefreshService<Channel, ClientConfiguration>>, QpuApiError> {
         let uri = parse_uri(address).map_err(QpuApiError::GrpcError)?;
-        let channel = get_channel(uri).map_err(|err| QpuApiError::GrpcError(err.into()))?;
+        let channel = get_channel_with_timeout(uri, self.timeout())
```

Comment:
As an earmark for a discussion elsewhere: seems like we should use a channel wrapper builder to support all the different options that might be applied to a channel and to prevent having to support every combination of options we might want

Reply:
Yeah, I agree. I added `get_channel_with_timeout` since it seemed like it would be one of our most common use cases and would be likely to save us some boiler plate. However, for future-proofing, I also added a `get_channel_with_endpoint` function that takes a tonic [`Endpoint`](https://docs.rs/tonic/latest/tonic/transport/channel/struct.Endpoint.html) that can be configured however a user likes.